### ISSUE TITLE
jflex has error-prone fall-through

### ISSUE THREAD

**Issue by regisd** [jflex-de/jflex#222]
jflex/jflex-unicode-maven-plugin/target/generated-sources/jflex/jflex/ArchaicLineBreakScanner.java:702: error: [FallThrough] Execution may fall through from the previous case; add a `// fall through` comment before this line if it was deliberate
```
          case 11: break;
          ^
```

see http://errorprone.info/bugpattern/FallThrough

**Comment by lsf37** [comment]
The fall-through violations seem to be in generated code -- we could automatically add the `// fall through` comment in the generator to indicate that this is on purpose.

**Comment by lsf37** [comment]
Right, that's actually what that specific fall through is for: to guard for the cases where there is no `return` or `throw` and the scan just continues. Do we get warnings for fall through comments that are spurious? If not, we could add another line with something like
`// allowed to fall through if action does not return or throw`

**Comment by wrprice** [comment]
Curious... why use fall through to a new case instead of adding an explicit `break;` ?

**Comment by regisd** [comment]
`// fall through` is the exact opposite of `break`. It means the case falls to the following case statement, and that the developer has not forgotten the `break`.

**Comment by wrprice** [comment]
I understand, but I noticed quite a few cases that fall-through to cases that _only_ `break;` and do nothing else.  I was just about to edit my post because the answer is probably because you don't know whether the action code included a return or break, and without examining it the safest thing is to fall-through to a separate case that breaks.  Blindly inserting a break after an action could result in an unreachable statement otherwise.

Sorry for the noise.

**Comment by regisd** [comment]
There are still violations 
https://travis-ci.org/jflex-de/jflex/jobs/431900782

**Comment by regisd** [comment]
Work around: Use javac option `-Xep:FallThrough:WARN`

**Comment by regisd** [comment]
I also believe that JFlex 1.7.0 has fall-through violations that 1.6.1 didn't have.
